File: java/src/main/java/net/razorvine/pyro/PyroException.java
package net.razorvine.pyro;

/**
 * Exception thrown when something is wrong in Pyro.
 *
 * @author Irmen de Jong (irmen@razorvine.net)
 */
public class PyroException extends RuntimeException {

	private static final long serialVersionUID = 5164514665621511957L;
	public String _pyroTraceback;
	public String pythonExceptionType;

	public PyroException() {
		super();
	}

	public PyroException(String message, Throwable cause) {
		super(message, cause);
	}

	public PyroException(String message) {
		super(message);
	}
}

File: dotnet/Razorvine.Pyrolite/Pyrolite/Pyro/PyroException.cs
/* part of Pyrolite, by Irmen de Jong (irmen@razorvine.net) */

using System;
using System.Runtime.Serialization;
// ReSharper disable InconsistentNaming

namespace Razorvine.Pyro
{
	/// <summary>
	/// Exception thrown when something is wrong in Pyro, or an exception was returned from a remote call.
	/// </summary>
	[Serializable]
	public class PyroException : Exception
	{
		public string _pyroTraceback {get; set;}
		public string PythonExceptionType {get; set;}

		public PyroException()
		{
		}

	 	public PyroException(string message) : base(message)
		{
		}

		public PyroException(string message, Exception innerException) : base(message, innerException)
		{
		}

		// This constructor is needed for serialization.
		protected PyroException(SerializationInfo info, StreamingContext context) : base(info, context)
		{
		}
	}
}
